exercises/clock/test/test_clock.c

```diff
@@ -0,0 +1,489 @@
+                                minute_offset);
+}
+
+void test_add_more_than_one_day(void)
+{
+   int hour = 5;
+   int minute = 32;
+   int minute_offset = 1500;
+   char *expected_clock_text = "06:32";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_add_more_than_two_days(void)
+{
+   int hour = 1;
+   int minute = 1;
+   int minute_offset = 3500;
+   char *expected_clock_text = "11:21";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_minutes(void)
+{
+   int hour = 10;
+   int minute = 3;
+   int minute_offset = -3;
+   char *expected_clock_text = "10:00";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_to_previous_hour(void)
+{
+   int hour = 10;
+   int minute = 3;
+   int minute_offset = -30;
+   char *expected_clock_text = "09:33";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_an_hour(void)
+{
+   int hour = 10;
+   int minute = 3;
+   int minute_offset = -70;
+   char *expected_clock_text = "08:53";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_across_midnight(void)
+{
+   int hour = 0;
+   int minute = 3;
+   int minute_offset = -4;
+   char *expected_clock_text = "23:59";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_two_hours(void)
+{
+   int hour = 0;
+   int minute = 0;
+   int minute_offset = -160;
+   char *expected_clock_text = "21:20";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_two_hours_with_borrow(void)
+{
+   int hour = 6;
+   int minute = 15;
+   int minute_offset = -160;
+   char *expected_clock_text = "03:35";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_one_day(void)
+{
+   int hour = 5;
+   int minute = 32;
+   int minute_offset = -1500;
+   char *expected_clock_text = "04:32";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+void test_subtract_more_than_two_days(void)
+{
+   int hour = 2;
+   int minute = 20;
+   int minute_offset = -3000;
+   char *expected_clock_text = "00:20";
+
+   test_time_add_expected_value(expected_clock_text, hour, minute,
+                                minute_offset);
+}
+
+// these tests check equality.
+// Part of standard test collection
+// Included even though they don't make much sense in C
```

Comment:
Agreed, these don't make much sense in C since a "clock" is just a string and I think we fairly conclusively proved these above.

However, they _could_ make sense if we were to allocate clock objects and provide a clock object comparison.

I'm okay if we don't refactor to add clock objects, but we should probably remove these if we don't since their inclusion is puzzling for the C implementation.